ui-participant/src/studies/enroll/StudyEnrollRouter.tsx

```diff
@@ -150,14 +151,36 @@ function StudyEnrollOutletMatched(props: StudyEnrollOutletMatchedProps) {
       } else {
         // when preEnroll is satisfied, and we have a user, we're clear to create an Enrollee
         try {
-          const hubUpdate = isProxyEnrollment
+          const hubResponse = isProxyEnrollment
             ? await enrollProxyUserInStudy(
-              studyShortcode, studyName, preEnrollResponseId, ppUserId, refreshLoginState, i18n
+              studyShortcode, preEnrollResponseId, ppUserId, refreshLoginState
             )
             : await enrollCurrentUserInStudy(
-              studyShortcode, studyName, preEnrollResponseId, refreshLoginState, i18n
+              studyShortcode, preEnrollResponseId, refreshLoginState
             )
-          navigate('/hub', { replace: true, state: hubUpdate })
+
+          const sortedActiveConsentTasks = getSortedActiveTasks(hubResponse.enrollee.participantTasks, 'CONSENT')
+          const nextConsentTask = getNextTask(hubResponse.enrollee, sortedActiveConsentTasks)
+
+          if (nextConsentTask) {
+            //if there's a pending consent, navigate directly to the first one. note that we don't
+            //show the "Welcome to the study" banner in this case
+            const consentTaskPath = getTaskPath(
+              nextConsentTask, hubResponse.enrollee.shortcode, studyShortcode)
+            navigate(`../../hub/${consentTaskPath}`, { replace: true })
+          } else {
+            const hubUpdate: HubUpdate = {
+              message: {
+                title: i18n('hubUpdateWelcomeToStudyTitle', {
+                  substitutions: { studyName }
+                }),
+                detail: i18n('hubUpdateWelcomeToStudyDetail'),
+                type: 'INFO'
+              }
+            }
+
```

Comment:
there's a lot in common here with the redirect.  Can we make a `handleNewStudyEnroll` function to consolidate?